packages/venia-ui/lib/policies/index.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Custom type policies that allow us to have more granular control
+ * over how ApolloClient reads from and writes to the cache.
+ *
+ * https://www.apollographql.com/docs/react/caching/cache-configuration/#typepolicy-fields
+ * https://www.apollographql.com/docs/react/caching/cache-field-behavior/
+ */
+const typePolicies = {
+    // Query/Mutation are "types" just like "Cart".
+    Query: {
+        fields: {
+            cart: {
+                // Replaces @connection(key: "Cart")
```

Comment:
Makes me think we should update our [Mutation PII Cheat Sheet](https://github.com/magento/pwa-studio/wiki/Project-coding-standards-and-conventions#mutation-pii-cheat-sheet).

Reply:
Not sure if the use of the policy for the `Cart` field warrants the update, but I think I get where you're going with this. We could preempt this and move all the connection directives from mutations to this policies file. For example this mutation:

```gql
export const REMOVE_GIFT_CARD_MUTATION = gql`
    mutation removeGiftCard($cartId: String!, $giftCardCode: String!) {
        removeGiftCardFromCart(
            input: { cart_id: $cartId, gift_card_code: $giftCardCode }
        ) @connection(key: "removeGiftCardFromCart") {
            ...
       }
    }
`;
```

would become

```js
{
   Mutation: {
       fields: {
           removeGiftCardFromCart: {
               keyArgs: () => 'removeGiftCardFromCart'
           }
       }
   }
}
```

Then the rule in the cheatsheet, and something we'd have to enforce in reviews, would be to add a field `keyArg` rule to the policies file when adding a new mutation.

I'd like to get opinions on that though.

Reply:
[![Image from Gyazo](https://i.gyazo.com/a0f2b9d247a5e04093b9afcaa389b81e.png)](https://gyazo.com/a0f2b9d247a5e04093b9afcaa389b81e)

Reply:
Look into `no-cache` global policy for mutations.